web/src/components/MapView.tsx

```diff
@@ -113,40 +102,58 @@ const MapView = React.forwardRef((props: MapViewProps, ref: React.Ref<MapRef>):
     }
   }, [currentFeature, featureCollection, queryLocation, queryParams, selectFeature])
 
-  const onSelectFeature = (e: MapEvent) => {
-    if (e.features?.length) {
-      flyToPoi(e.features[0].geometry.coordinates)
-      selectFeature(e.features[0])
-      changeSnapPoint(1)
-      queryParams.set(locationName, e.features[0].properties.urlSlug)
-      updateQueryParams(queryParams)
-    } else {
-      setQueryLocation(null)
-      selectFeature(null)
-      updateQueryParams()
-    }
-  }
+  const onDeselectFeature = useCallback(() => {
+    setQueryLocation(null)
+    selectFeature(null)
+    updateQueryParams()
+  }, [selectFeature, setQueryLocation])
+
+  const onSelectFeature = useCallback(
+    event => {
+      // Since the event is only fired if canvas on layer was clicked, the container propagation has to be stopped on deselect
+      event.originalEvent.stopPropagation()
+      const feature = event.features && event.features[0]
+      if (feature) {
+        flyToPoi(feature.geometry.coordinates)
+        selectFeature(feature)
+        changeSnapPoint(1)
+        queryParams.set(locationName, feature.properties?.urlSlug)
+        updateQueryParams(queryParams)
```

Comment:
Why do we need to set query params twice?

Reply:
first time you set it in the queryParams object and then you push it to the history. if you don't push it nothing happens with the object